### PR TITLE
Rework replication module docs

### PR DESCRIPTION
**Pull request by wallnerryan**
# Description
Consolidate duplicate table data, add details on file/block sync/async for replication module. Make landing page tables linkable to module specific docs. 

# GitHub Issues
None

# Checklist:

- [X] Have you run a grammar and spell checks against your submission?
- [X] Have you tested the changes locally?
- [X] Have you tested whether the hyperlinks are working properly?
- [N/A] Did you add the examples wherever applicable?
- [N/A] Have you added high-resolution images?

### CODE REVIEW

**Comment by wallnerryan** [comment]
@shanmydell after speaking with @coulof about a few user conversations where this info was not readily available we thought it would be good to clean up and make additions to clear up replications support across file/block and sync/async. I noticed there was an existing table with dome duplicate data, so I consolidated to one as well.

**Review by rsedlock1958 — DISMISSED** [review]
I have reviewed this. No edits are required.

File: content/docs/replication/_index.md [hunk 1]
```diff
@@ -19,13 +19,20 @@ CSM for Replication provides the following capabilities:
 | Capability                                                                          | PowerMax | PowerStore | PowerScale | PowerFlex | Unity |
 | ----------------------------------------------------------------------------------- | :------: | :--------: | :--------: | :-------: | :---: |
 | Replicate data using native storage array based replication                         | yes      | yes        | yes        | no        | no    |
+| Asynchronous file volume replication                         | yes      | yes        | yes        | no        | no    |
```
Comment (coulof) [inline content/docs/replication/_index.md:22]: To be confirmed by the engineering team, I think we do async block (yes) and not async file (no).

File: content/docs/replication/_index.md [hunk 2]
```diff
@@ -19,13 +19,20 @@ CSM for Replication provides the following capabilities:
...
+| Asynchronous block volume replication                         | yes      | no        | n/a        | no        | no    |
+| Synchronous file volume replication                         | yes      | no        | no        | no        | no    |
+| Synchronous block volume replication                         | yes      | no        | n/a        | no        | no    |
+| Active-Active (Metro) block volume replication                         | yes      | no        | n/a        | no        | no    |
+| Active-Active (Metro) file volume replication                         | yes      | no        | no        | no        | no    |
 | Create `PersistentVolume` objects in the cluster representing the replicated volume | yes      | yes        | yes        | no        | no    |
 | Create `DellCSIReplicationGroup` objects in the cluster                             | yes      | yes        | yes        | no        | no    |
 | Failover & Reprotect applications using the replicated volumes                      | yes      | yes        | yes        | no        | no    |
 | Online Volume Expansion for replicated volumes				       | yes      | no	      | no	   | no	       | no    |
```
Comment (coulof) [inline content/docs/replication/_index.md:31]: @ChristianAtDell, do you know if we have a ticket to support resize with replicated PV ?

File: content/docs/replication/_index.md (hunk 1 above)
Comment (wallnerryan) [reply content/docs/replication/_index.md:22]: Ah right, I updated the spreadsheet but not here.  Will update this.

Comment (wallnerryan) [reply content/docs/replication/_index.md]: this should be updated with latest commit

**Review by coulof — APPROVED** [review]
LGTM

**Review by shanmydell — APPROVED** [review]

**Pull request merged**